### ISSUE TITLE
Handle Esc key for multiple components

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1051]
There are several components that should be dismissible via the Esc key:

- Modal
- Popover
- Dropdown when expanded
- Tooltip

Some of these components can be nested, e.g. a modal with a dropdown or tooltip. For these cases, the Esc key should not dismiss everything that's open. Instead, the Esc key should only dismiss the last component that was expanded/opened.

As part of this issue, we need provide one of the following: 
- ability to manage the Esc key behavior for multiple components
- a POC that demonstrates this ability, documentation on how it was achieved, and a new issue to capture how we want to share this information with consumers

**Comment by jessiehuff** [comment]
@jgiardino I was taking a look at these components, and I think they all currently support Esc key interaction. I think you make a valuable point about nested components, do you think that part of the issue is related to this [issue](https://github.com/patternfly/patternfly-react/issues/3397)?
